explainer.md

```diff
@@ -0,0 +1,802 @@
+
+These objects all have in common is that they are not tied to a specific point on the timeline.
+
+### CivilDate
+
+`CivilDate` (and its siblings) represents a date and time corresponding to the requirements of ISO-8601.
+
+This means specifically that it uses the *proleptic Gregorian calendar* whose days are 24 hours
+long and begin and end at midnight.
+
+#### CivilDate() - constructor
+
+The constructor may only be called as such. It takes 3 numeric arguments.
+
+- `year` the Gregorian year
+- `month` the Gregorian month
+- `day` the Gregorian day of the month
+
+#### CivilDate.prototype.year : number
+
+The `.year` property represents the year of the `CivilDate`.
+
+#### CivilDate.prototype.month : number
+
+The `.month` property represents the month of the `CivilDate`.
+
+#### CivilDate.prototype.day : number
+
+The `.day` property represents the day of the month of the `CivilDate`.
+
+#### CivilDate.prototype.dayOfWeek : number
+
+The `.dayOfWeek` property represents the day of the week where Monday is `1` and `Sunday` is `7` in accordance with ISO-8601.
+
+#### CivilDate.prototype.dayOfYear : number
+
+The `.dayOfYear` property represents the ordinal day of the Gregorian year according to ISO-8601.
+
+#### CivilDate.prototype.weekOfYear : number
+
+The `.weekOfYear` property represents the ISO week-number. Beware that dates at the begining of a year may be part of a week from the preceding year, and dates at the end of a year may be part of a week at the beginning of the next year, as the first week of any year is defined as the week that contains the first Thursday of the week.
+
+#### CivilDate.prototype.plus(value: IntervalLike) : CivilDate
+
+Creates a new `CivilDate` object by adding (subtracting for negative values) values to its members.
+The specified values must be numeric if specified.
+
+The algorithm is such that:
+  1. the individual values are added to the existing values.
+  2. the range of `days` is ensured to be between 1 and 29-31 depending on the month by adjusting `month`
+  3. the range of `months` is ensured to be between 1 and 12 by adjusting the `years`.
+
+#### CivilDate.prototype.with(values: DateLike) : CivilDate
+
+Creates a new `CivilDate` object by overriding specified values to its members.
+The specified values must be numeric if specified.
+
+#### CivilDate.prototype.withTime(time : CivilTime) : CivilDateTime
+
+Combines this `CivilDate` with the passed `CivilTime` to create a new `CivilDateTime` object.
+
+#### CivilDate.prototype.toString() : string
+
+Equivalent to `date.toDateString()`
+
+#### CivilDate.prototype.toJSON() : string
+
+Equivalent to `date.toString()`
+
+#### CivilDate.prototype.toDateTimeString() : string
+
+`.toDateString()` creates an ISO-8601 compliant string in the format:
+**`year`-`month`-`day`**. The `year` is 0-padded to a minimum of 4 digits. `month` and `day`.
```

Comment:
What is the purpose of a method on CivilDate whose name suggests that it depends upon or produces time data?

Reply:
It's an oversight and should be `toDateString()`